### PR TITLE
WIP: XCode 8 Compatibility

### PR DESCRIPTION
**Pull request by LorenzMeier**
XCode 8 improves the POSIX compatibility of Mac OS and we take advantage of it and rip out some hackery we had been doing to make it compatible.

### CODE REVIEW

File: cmake/configs/posix_sitl_default.cmake [hunk 1]
```diff
@@ -88,7 +88,7 @@ set(config_module_list
 	#modules/mavlink/mavlink_tests #TODO: fix mavlink_tests
 	modules/unit_test
 	modules/uORB/uORB_tests
-	systemcmds/tests
+	#systemcmds/tests
```
Comment (LorenzMeier) [inline cmake/configs/posix_sitl_default.cmake]: This needs to be added back in once fixed.

File: Makefile [hunk 2]
```diff
@@ -261,14 +261,14 @@ package_firmware:
...
-	@$(MAKE) -C NuttX/nuttx clean
+	-@$(MAKE) -C NuttX/nuttx clean
 
 submodulesclean:
 	@git submodule sync --recursive
 	@git submodule deinit -f .
 	@git submodule update --init --recursive --force
 
-distclean: submodulesclean
+distclean: submodulesclean clean
```
Comment (dagar) [inline Makefile]: Is this needed? If you have a completely fresh repo does nuttx clean change anything?

Comment (LorenzMeier) [inline Makefile]: We need to wipe the build folders, which otherwise was missing.

**Comment by julianoes** [comment]
Looks correct to me.
Tested on macOS 10.11 with XCode 8.0.

File: Makefile (hunk 2 above)
Comment (dagar) [inline Makefile]: The git clean below wipes out everything except the eclipse project files. Unless it isn't for you?

**Pull request merged**